src/ebonite/repository/metadata/local.py

```diff
@@ -252,8 +261,10 @@ def update_project(self, project: Project) -> Project:
         return project
 
     def delete_project(self, project: Project):
+        if project.tasks:
```

Comment:
I recommend using `self.get_tasks` here as `Project.tasks` is not guaranteed to reflect DB state.